### ISSUE TITLE
Invoice on Account/Advance Invoice

### ISSUE THREAD

**Issue by marination** [frappe/erpnext#34467]
## Proposal:
In ERPNext, all Sales Invoice behave like a Partial/Final Invoice, meaning that they sell actual quantities of an item or service at a certain rate.

What we need in addition to Sales Invoice, is something called Invoice on Account. This is basically just a document telling the customer “for all the (unfinished) work I did so far, please pay me a lump sum X in advance.” The receipt of this amount should be recorded as an **advance payment against the Sales Order**. 
After the job (or some clearly separable part of it) is finished, the user crates a Sales Invoice (Partial/Final) and fetches any advance payments to be deducted from the total.

Payment Entry against the Sales Order should be created from the Invoice On Account.

So the flow for advance payment could be (optionally): **Sales Order** > **Invoice On Account** > **Payment Entry**

All craftsmen in Germany currently need this.
This can also be fashioned as an Advance Receipt Voucher in India perhaps (https://www.teachoo.com/6361/1947/What-is-Receipt-Voucher-in-GST/category/GST-Invoice-Format/) with regional custom fields for GSTIN, etc.

Ref: https://www.deskera.com/blog/advance-invoice/

![screencapture-dach-dev-8001-app-invoice-on-account-ACC-INVACC-2023-00002-2023-03-16-12_09_41](https://user-images.githubusercontent.com/25857446/225536111-a6e26e47-b5a1-4570-8895-0011f7b24002.png)

## Alternatives considered:
**Proforma Invoice** does not work with Lump Sum amounts (item and qty is required), neither does it show exactly how much advance is being requested

## Further Reading

- https://learning.sap.com/learning-journeys/implement-sap-s-4hana-cloud-public-edition-for-professional-services/billing-on-account-and-down-payments_dc2e8628-393b-4003-8027-7ba25e0e3908
- https://learn.microsoft.com/en-us/dynamicsax-2012/appuser-itpro/about-on-account-invoicing
- https://www.odoo.com/documentation/16.0/applications/sales/sales/invoicing/down_payment.html
- https://apps.datev.de/help-center/documents/1070732 (german)
- https://apps.datev.de/help-center/documents/9211369 (german)
- https://apps.datev.de/help-center/documents/1070732 (german)

Related issue: https://github.com/frappe/erpnext/issues/42843

**Comment by digitalrevisor** [comment]
Important feature for all business as well

**Comment by nakul80** [comment]
Very Important to get this feature. If I give a service PO, its almost guaranteed that there would be RA (running ac) bills. These are tax invoices requiring to be registered and accounted for taxation setoff etc. System doesnt allow this. I tried to set new UOM called Job which doesnt have to be a whole number. This way I can register supplier invoice partially to achieve this result. It works but not the cleanest method.

**Comment by kittiu** [comment]
Is this valid? https://github.com/ecosoft-frappe/erpnext_thailand/pull/17

**Comment by nakul80** [comment]
It might work but I dont see it on my site. 

Do I have to install it? @kittiu

**Comment by kittiu** [comment]
> It might work but I dont see it on my site.
> 
> Do I have to install it? [@kittiu](https://github.com/kittiu)

Yes, you need to install erpnext_thailand to use.

**Comment by NikoKS** [comment]
Really need this feature as a feature in ERPNext

**Comment by worcworc** [comment]
+1

**Comment by DipenFrappe** [comment]
@marination, please comment on how the accounting will happen in all the steps

**Comment by kittiu** [comment]
Yes, ERPNext will still need this feature badly. In fact, I got another customer that has multiple deposit invoice not just once as we try to do in https://github.com/ecosoft-frappe/erpnext_thailand/pull/17

cc @newtratip

**Comment by DipenFrappe** [comment]
@kittiu @marination, help with the accounting impacts on every step. For example, if you pay taxes on payment and then create a sales invoice, again taxes will be applied on the sales invoice, so how are taxes expected to be applied and how accounting will be happening. Also consider partial, multiple payments, and one invoice, etc.

**Comment by HenningWendtland** [comment]
@DipenFrappe I have this developed for german needs in a currently private repo on v15.
Workflow is exactly as in the issue. Problem on the accounting side is that ERPNext does not handle advance taxes correct/well, so my Payment Entry cannot use them, see [42843](https://github.com/frappe/erpnext/issues/42843)

### Some more context: 
In Germany (and other countries according to a quick AI search) tax laws state that taxes on advances are only due upon receipt of payment. In financial accounting also no revenue can be recorded when the goods/services have not been delivered. So submitting an "Advance Invoice" should actually **not affect the books**. Only a corresponding Payment Entry should do that.

**Implication**: An **Invoice on Account / Advance Invoice** Doctype nicely fits in the ERPNext Advance Process, since itself does not affect accounting. The main issue from my point of view is that the Payment Entry / regular SI does not resolve advance taxes correctly, which should be recorded in the PE

**Comment by 0xD0M1M0** [comment]
> For example, if you pay taxes on payment and then create a sales invoice, again taxes will be applied on the sales invoice, so how are taxes expected to be applied and how accounting will be happening. Also consider partial, multiple payments, and one invoice, etc.

From what I see during my investigations on this topic, is that the Payment Entry can handle the tax related topic for the Advance Payment quite well ("Book Advance Payments in Separate Party Account" in company settings enabled and "Default Advance Received Account" set). If the advance payment then is linked with the sales invoice, the amount without tax is transferred correctly from the Advance Received Account to the Payable Account, but the "Advance Tax" (which is correctly booked by the payment entry in the GL), is not transferred to the Payable Account and therefore the tax is currently double.

Looking at the topic closer reveals more issues for solving, like "which "Advance Taxes and Charges" amount belongs to which "Payment References", so it is resolved on booking the Sales Invoice correctly.

**Comment by barredterra** [comment]
In case we decide against a dedicated DocType, I think we could achieve the same functionality with three additional settings for Sales Invoices:

1. Allow items on **Sales Invoice** to differ from **Sales Order** (so we can bill a lump sum using some generic item)
2. A "Select" field _Taxes Due On_ with values "Time of Invoice" and "Time of Payment" (steers posting of tax amounts)
3. A "Check" field _Services Rendered_ (steers posting of either revenue for invoice or liability for payment)

**Comment by venkatamandavilli-code** [comment]
This issue hits very close to home. In service‑based businesses, it’s completely normal to bill a client mid‑project for work completed so far — without breaking everything down into line items or quantities. That’s simply not how service billing works.

Right now, ERPNext forces every invoice to be tied to items and quantities. That’s fine for product sales, but it feels unnatural and restrictive for service delivery. Workarounds like creating a dummy UOM such as “Job” or relying on Proforma Invoices only add confusion for clients and accountants.

The proposed flow — **Sales Order → Invoice on Account → Payment Entry → Final Sales Invoice with advance deduction** — is clean, practical, and aligns with what mature ERPs like SAP, Odoo, and Microsoft Dynamics have supported for years.

And this isn’t just a Germany‑specific requirement. Any project‑based or service‑driven business anywhere in the world needs this capability. Really hoping the core team prioritizes it.

**Comment by vorasmit** [comment]
Nothing against the feature `on account invoice`.

Fact check 
-> `on account invoice` != `advance payment`.

My understanding:
- on account invoice is created after milestones / as per payment terms agreed
- items could be differently described
- taxes are immediately due for such invoice (in India) and should be paid - not on receipt of payment
- final invoice is of balance amount - can include references to on account invoices for customers satisfaction, but is not a requirement for compliance, as invoices are individually reported

So this should be probably supported as part of Sales Invoice itself.

validated this from: https://www.odoo.com/documentation/19.0/applications/sales/sales/invoicing/down_payment.html

**Comment by HenningWendtland** [comment]
> Fact check -> `on account invoice` != `advance payment`.

Yes these are two different things. I think the terminologies and tax laws are different in India / Germany, which makes it confusing. But in principal we have two things, both valid and potentially needed:

- **Advance Invoice** (in the Issue description called Invoice on Account): **without Tax / Accounting implication**. Here the issue is more to make ERPNext handle taxes on advance receipt properly, see https://github.com/frappe/erpnext/pull/54921. While I do not know much about Indian laws, a quick [research](https://finaxis.in/gst-on-advance-payment/#:~:text=For%20example%2C%20if%20a%20company,even%20before%20completing%20the%20service.)  indicates that you would also need this, since GST on Advances seems due upon receipt of the Advance normally. The "Advance Invoice" Doctype could then sandwich between SO and PE, like in the Issue description. Complexity is less, since books are not affected
- **Invoice on Account** (in your sense above): with Tax and Accounting Implication and potentially different items table, so this would be part of the SI and is a different topic but still valid

About the Terminology: In Germany companies (as buyers) often require a formally correct invoice asking for an Advance Payment before they pay out the Advance. However that invoice has no accounting / tax implications for the seller until the money is received. The sellers books are affected and GST is due when he records the advance. So comparing with India, it seems like a bit in between a Proforma Invoice (formless, non-binding, asking for an andvance) and an On Account Invoice (full invoice with accounts affected and taxes due), but probably better implemented outside of SI, since accounting is not affected.
